### PR TITLE
Career resource api

### PR DESCRIPTION
**Pull request by SundayAdefila**
Implemented the career routes with full spec support.

@ceemion  kindly check at earliest possible convenience.

### CODE REVIEW

File: app/controllers/api/v1/careers_controller.rb [hunk 1]
```diff
@@ -0,0 +1,62 @@
+class Api::V1::CareersController < Api::ApiController
+
+  before_action :set_api_v1_career, except: [:index, :create]
+
+  def index
+    @api_v1_careers = Career.all
```
Comment (ceemion) [inline app/controllers/api/v1/careers_controller.rb]: Do we need any form of backend order/filters, or should we do all in the front-end?
I suggest you return careers with the last updated first, something like this `Career.order('updated_at': 'DESC')`.

File: spec/models/career_spec.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
+require 'rails_helper'
+
+RSpec.describe Career, type: :model do
+  it "should have a valid factory" do
+    career = FactoryGirl.build(:career)
+    expect(career).to be_valid
+  end
+
```
Comment (ceemion) [inline spec/models/career_spec.rb]: You may consider making this test more robust to cover more scenarios and edge cases.

File: app/controllers/api/v1/careers_controller.rb (hunk 1 above)
Comment (SundayAdefila) [inline app/controllers/api/v1/careers_controller.rb]: @ceemion : Yes, we will need order. I plan to implement pagination later. so, the api will take in params like "from" and "to". Right now, I'm wiring up the basics for the sake of the joins...

Comment (ceemion) [inline app/controllers/api/v1/careers_controller.rb]: Ok, that's cool.
